### PR TITLE
add domain and method aware logging to connectionpool

### PR DESCRIPTION
**Pull request by birm**
One of my attempts at #881. I would consider at most one of the two (#898).

### CODE REVIEW

**Comment by birm** [comment]
I don't really understand the pypy error here.

**Comment by Lukasa** [comment]
The PyPy error is now magically gone. =D

I'd be happy with this change. Up to @shazow though!

**Comment by haikuginger** [comment]
Looks good to me!

**Comment by shazow** [comment]
Should we be including the scheme too, while we're at it?

**Comment by haikuginger** [comment]
I vote yes for including scheme; wanting to know if your requests are happening securely is a valid use case for logging. But it should be based on the calling class scheme, rather than the URL's scheme; as seen in #869, the URL passed can be incorrect if `assert_same_host` is passed as False.

**Comment by birm** [comment]
Okay, I agree, so I added the scheme. I'm not sure if the way I wrote it could be better though. thoughts?

**Comment by Lukasa** [comment]
I think I'd like to skip `using scheme` if possible: it's nice to have logs that are relatively structured (e.g. can be parsed by grep). Either we should just find a place in there to put the string, or we should consider using `key=value` logic for some of it.

File: urllib3/connectionpool.py [hunk 1]
```diff
@@ -397,8 +397,9 @@ def _make_request(self, conn, method, url, timeout=_Default, chunked=False,
 
         # AppEngine doesn't have a version attr.
         http_version = getattr(conn, '_http_vsn_str', 'HTTP/?')
-        log.debug("\"%s %s %s\" %s %s", method, url, http_version,
-                  httplib_response.status, httplib_response.length)
+        log.debug("%s:%s \"%s %s %s\" %s %s using scheme %s ", self.host, self.port,
```
Comment (shazow) [inline urllib3/connectionpool.py]: Would `https://127.0.0.1:8000` be better?

Comment (birm) [inline urllib3/connectionpool.py]: That's much better, and I should have done that from the start.

Comment (shazow) [inline urllib3/connectionpool.py]: 👍 no worries. :)

**Comment by birm** [comment]
I now have scheme before the host; (scheme)://(host):(port) ...
Does this break anyone's workflow?

**Comment by shazow** [comment]
Only one way to find out. :D

**Pull request merged**